includes/Core/Util/User_Input_Settings.php

```diff
@@ -126,19 +126,27 @@ function( $setting ) {
 	 * @return array|WP_Error User input settings.
 	 */
 	private function sync_with_proxy( $settings = null ) {
-		$user_input_settings_url  = $this->authentication->get_google_proxy()->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
+		$user_input_settings_url = $this->authentication->get_google_proxy()->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
+		$creds                   = $this->authentication->credentials()->get();
+
 		$user_input_settings_args = array(
 			'headers' => array(
 				'Authorization' => 'Bearer ' . $this->authentication->get_oauth_client()->get_access_token(),
+				'Content-Type'  => 'application/json',
+			),
+			'body'    => array(
```

Comment:
The body should be JSON encoded here. Eg. `'body'    => wp_json_encode( array(`

Reply:
Thanks for these comments adam, I've worked down each below.